### DISCUSSION TITLE
Can I use spectre.console to print to my own colored console?

### DISCUSSION THREAD

**Discussion by goswinr** [spectreconsole/spectre.console#573] [Q&A] [answered]
My WPF app has its own console (based on Avalonedit).
For colored printing there is a function that takes three integers and a string:
`PrintColored(red, green, blue, text)`
Can I use spectre.console to print in color to my own console?

**Comment by patriksvensson** [comment, marked as answer]
@goswinr If your console exposes a `TextWriter` and supports ANSI escape sequences then yes.

**Comment by goswinr** [comment]
@patriksvensson yes my console exposes a` TextWriter ` too. 
Would I have to parse the ANSI escape sequences myself within the` TextWriter`  to call my `PrintColored(red, green, blue, text)` ?

**Comment by patriksvensson** [comment]
@goswinr Exactly. If you do that you should be able to use Spectre.Console in your app.

**Comment by goswinr** [comment]
@patriksvensson Thanks, it works already great in my F# scripting editor. Just the colors are missing.
![2020-10-21 12_31_51](https://user-images.githubusercontent.com/884885/96708405-7d6ab000-1399-11eb-9ed6-239fc18ceb8b.png)

**Comment by patriksvensson** [comment]
@goswinr Awesome! Your app looks awesome as well! 😀
